libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline/validations/relations.rs

```diff
@@ -509,9 +503,10 @@ pub(super) fn referencing_scalar_field_types(relation: InlineRelationWalker<'_,
     }
 }
 
-fn is_empty_fields(fields: Option<&[ast::FieldId]>) -> bool {
+fn is_empty_fields<T>(fields: Option<impl ExactSizeIterator<Item = T>>) -> bool {
     match fields {
-        None | Some([]) => true,
+        None => true,
+        Some(fields) if fields.len() == 0 => true,
```

Comment:
Couldn't that just be Some(fields)=> fields.len() == 0 ?

Reply:
absolutely, will do